src/Sylius/Bundle/CoreBundle/Controller/OrderController.php

```diff
@@ -24,6 +28,76 @@ class OrderController extends BaseOrderController
      *
      * @return Response
      */
+    public function completeAction(Request $request)
+    {
+        $configuration = $this->requestConfigurationFactory->create($this->metadata, $request);
+
+        $this->isGrantedOr403($configuration, ResourceActions::UPDATE);
+        $order = $this->findOr404($configuration);
+
+        $form = $this->resourceFormFactory->create($configuration, $order);
+
+        if (in_array($request->getMethod(), ['POST', 'PUT', 'PATCH'], true) && $form->handleRequest($request)->isValid()) {
+            $order = $form->getData();
+
+            $event = $this->eventDispatcher->dispatchPreEvent(ResourceActions::UPDATE, $configuration, $order);
+
+            if ($event->isStopped() && !$configuration->isHtmlRequest()) {
+                throw new HttpException($event->getErrorCode(), $event->getMessage());
+            }
+            if ($event->isStopped()) {
+                $this->flashHelper->addFlashFromEvent($configuration, $event);
+
+                return $this->redirectHandler->redirectToResource($configuration, $order);
+            }
+
+            try {
+                if ($configuration->hasStateMachine()) {
+                    $this->stateMachine->apply($configuration, $order);
+                }
+
+                $this->manager->flush();
+            } catch (OptimisticLockException $exception) {
+                $this->addFlash('error', 'sylius.checkout.complete_error');
+
+                return $this->redirectHandler->redirectToRoute($configuration, 'sylius_shop_homepage');
+            }
+
+            $this->eventDispatcher->dispatchPostEvent(ResourceActions::UPDATE, $configuration, $order);
+
+            $this->getOrderEmailManager()->sendConfirmationEmail($order);
```

Comment:
I was looking for this and also checked when emails are sent, but there are sent immediately after call `send` method, so I couldn't clear the email queue. 
In our test environment we are using this approach: http://symfony.com/doc/current/email/spool.html
but in prod env it doesn't make sense.